app/models/notification.rb

```diff
@@ -83,13 +83,17 @@ def backup_subject_name
   def text_bindings
     [:initiator, :subject, :cause].each_with_object(text_bindings_dynamic) do |key, result|
       value = public_send(key)
+      next unless value
+
+      # Set the link based on the notification_type.link_to
+      result[:link] = {
+        :id    => value.id,
+        :model => value.class.name
+      } if notification_type.link_to.try(:to_sym) == key
+
       result[key] = {
-        :link => {
```

Comment:
I think this is okay, but I just wanted to call out what looks like a slight change in behavior.

Because you are moving which key to link to to the notification type, the `:link` key is getting moved up as a sibling to the relation keys.

Prior to this change it was possible to set a link for each of `:initiator`, `:subject` and, `:cause`, but now we can only set it to one as determined by the notification type.

Reply:
Yup, the design here was for up to 3 links for a notification with the links in the notification text itself, however, UX designed the visual part differently so only one link is allowed.